src/main/java/bio/ferlab/fhir/schema/definition/specificity/ReferenceDefinition.java

```diff
@@ -11,12 +13,17 @@
 
     @Override
     public JsonObject convertToJson(String root, String name, boolean required) {
-        JsonArray fields = Json.createArrayBuilder()
-                .add(JsonObjectUtils.createConst("reference", Constant.STRING, false))
-                .add(JsonObjectUtils.createConst("type", Constant.STRING, false))
-                .add(JsonObjectUtils.createConst("identifier", Constant.STRING, false))
-                .add(JsonObjectUtils.createConst("display", Constant.STRING, false))
-                .build();
-        return JsonObjectUtils.createInnerRecord(name, name, "A Reference", fields, required);
+        String formattedName = WordUtils.capitalize(name);
+        if (DefinitionRepository.registerInnerRecords(root, name)) {
+            return JsonObjectUtils.createRedefinedRecord(name, formattedName, Json.createObjectBuilder().build());
```

Comment:
A reference is also enforced by the namespace.fullname rule of Avro, so fixed that.